### PR TITLE
Fix disabled formfield validation

### PR DESCRIPTION
**Pull request by jslavitz**
Adds final polish to https://github.com/flutter/flutter/pull/19124 originally created by @gitvadim.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/form.dart [hunk 1]
```diff
@@ -256,6 +257,11 @@ class FormField<T> extends StatefulWidget {
   /// autovalidates, this value will be ignored.
   final bool autovalidate;
 
+  /// If true, indicates that the widget is enabled and if autovalidation is true
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/widgets/form.dart]: First I would describe what this does, then I would follow up with the details about autovalidation:

```suggestion
  /// Whether the widget/form? is able to receive user input.
  ///
  /// Defaults to true. If [autovalidation] is also true, the field will be validated if enabled
  /// is also true. Likewise, if enabled is false the field will not be validated regardless of
  /// [autovalidate]
  ///
```

File: packages/flutter/test/material/text_form_field_test.dart [hunk 2]
```diff
@@ -114,4 +114,52 @@ void main() {
...
+              validator: (String value) { _validateCalled += 1; return null; },
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(_validateCalled, 0);
+    await tester.showKeyboard(find.byType(TextField));
+    await tester.enterText(find.byType(TextField), 'a');
```
Comment (jonahwilliams) [inline packages/flutter/test/material/text_form_field_test.dart]: Does this work even when the form is disabled?

File: packages/flutter/lib/src/widgets/form.dart (hunk 1 above)
Comment (jslavitz) [reply packages/flutter/lib/src/widgets/form.dart]: Much better worded than me! Thanks!

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**